### PR TITLE
Update libddwaf to 1.13.1

### PR DESCRIPTION
**Pull request by iunanua**
### What does this PR do?
Update libddwaf to 1.13.1:
- replacing `ddwaf_object_signed` calls with `ddwaf_object_string_from_signed` [#interface-changes](https://github.com/DataDog/libddwaf/blob/master/UPGRADING.md#interface-changes)

Update libddwaf to 1.11.0:
- replacing `ddwaf_ruleset_info` with `ddwaf_object` and removing unused ruleset_info related methods [#ruleset-parsing-diagnostics](https://github.com/DataDog/libddwaf/blob/master/UPGRADING.md#ruleset-parsing-diagnostics)
- updating `DDWAF::init` and `DDWAF:update` returned object due changes in `ddwaf_result` [#ruleset-parsing-diagnostics](https://github.com/DataDog/libddwaf/blob/master/UPGRADING.md#ruleset-parsing-diagnostics)
- replacing `rulesInfo` property with `diagnostics` property [#ruleset-parsing-diagnostics](https://github.com/DataDog/libddwaf/blob/master/UPGRADING.md#ruleset-parsing-diagnostics)
- renaming `data` property as `events` [#events--actions-as-ddwaf_object](https://github.com/DataDog/libddwaf/blob/master/UPGRADING.md#events--actions-as-ddwaf_object)
- changing the type of `events` and `actions` properties to object [#events--actions-as-ddwaf_object](https://github.com/DataDog/libddwaf/blob/master/UPGRADING.md#events--actions-as-ddwaf_object)

### Checklist

- [ ] The CHANGELOG.md has been updated
- [ ] Unit tests have been updated and pass
- [ ] If known, an appropriate milestone has been selected

### CODE REVIEW

**Review by hoolioh — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "@datadog/native-appsec",
-  "version": "3.2.0",
+  "version": "3.3.0",
```
Comment (simon-id) [inline package.json:3]: Can we leave out the version bump out of this PR ?

Comment (iunanua) [reply package.json]: pushed

File: src/main.cpp [hunk 2]
```diff
@@ -292,12 +292,8 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
...
   if (code == DDWAF_MATCH) {
     res.Set("status", Napi::String::New(env, "match"));
-    res.Set("data", Napi::String::New(env, result.data));
-    Napi::Array actions = Napi::Array::New(env, result.actions.size);
-    for (uint32_t i = 0; i < result.actions.size; ++i) {
-      actions[i] = Napi::String::New(env, result.actions.array[i]);
-    }
-    res.Set("actions", actions);
+    res.Set("events", from_ddwaf_object(&result.events, env));
+    res.Set("actions", from_ddwaf_object(&result.actions, env));
```
Comment (iunanua) [inline src/main.cpp:296]: Are there `events` and `actions` only when there has been a match?

**Comment by simon-id** [comment]
In the new versions, libddwaf released builds for arm7 and linux x86, which we didn't support until now. APM was not too happy about that because dd-trace is supposed to support them. So it would be nice if we could include those now. You can try by removing this line: https://github.com/DataDog/dd-native-appsec-js/blob/main/.github/workflows/build.yml#L33

**Comment by simon-id** [comment]
You're missing `ddwaf_result.derivatives`: https://github.com/DataDog/libddwaf/blob/1.13.1/include/ddwaf.h#L168C18-L168C29

**Comment by iunanua** [comment]
> You're missing `ddwaf_result.derivatives`: https://github.com/DataDog/libddwaf/blob/1.13.1/include/ddwaf.h#L168C18-L168C29

As we are not using preprocessors I decided to keep it simple and not include the `derivatives` field...

File: package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "version": "3.2.0",
   "description": "Node.js bindings for libddwaf",
   "main": "index.js",
-  "libddwaf_version": "1.10.0",
+  "libddwaf_version": "1.13.1",
```
Comment (simon-id) [inline package.json:6]: update to new waf

Comment (iunanua) [reply package.json:6]: 1.14.0?

**Review by simon-id — APPROVED** [review]

File: src/main.cpp (hunk 2 above)
Comment (simon-id) [reply src/main.cpp:296]: Yes confirmed by Anil

**Pull request merged**